modules/dnn/src/dnn_utils.cpp

```diff
@@ -34,8 +76,13 @@ void blobFromImage(InputArray image, OutputArray blob, double scalefactor,
         const Size& size, const Scalar& mean, bool swapRB, bool crop, int ddepth)
 {
     CV_TRACE_FUNCTION();
-    std::vector<Mat> images(1, image.getMat());
-    blobFromImages(images, blob, scalefactor, size, mean, swapRB, crop, ddepth);
+    if (image.kind() == _InputArray::UMAT) {
+        std::vector<UMat> images(1, image.getUMat());
+        blobFromImages(images, blob, scalefactor, size, mean, swapRB, crop, ddepth);
+    } else if (image.kind() == _InputArray::MAT) {
+        std::vector<Mat> images(1, image.getMat());
+        blobFromImages(images, blob, scalefactor, size, mean, swapRB, crop, ddepth);
+    }
```

Comment:
`else` do NOOP - this is wrong.

Only `UMAT` and `STD_VECTOR_UMAT` should be handled in dedicated code path.
All other handling should not be changed by this patch.

Reply:
I understand

Reply:
[86095e6](https://github.com/opencv/opencv/pull/23894/commits/86095e6fb3ea563e643b9f3496dbaaedaeac0d57)